invoicing/models.py

```diff
@@ -218,31 +221,49 @@ def get_corresponding_apartment_installment(self, apartment_data):
         for field_name in field_names:
             setattr(apartment_installment, field_name, getattr(self, field_name))
 
-        if self.unit == InstallmentUnit.PERCENT:
-            if not self.percentage_specifier:
-                raise ValueError(
-                    f"Cannot calculate apartment installment value, {self} "
-                    f"has no percentage_specifier"
-                )
-            if self.percentage_specifier == InstallmentPercentageSpecifier.SALES_PRICE:
-                price_in_cents = apartment_data["sales_price"]
-            elif (
-                self.percentage_specifier
-                == InstallmentPercentageSpecifier.RIGHT_OF_OCCUPANCY_PAYMENT
-            ):
-                price_in_cents = apartment_data["right_of_occupancy_payment"]
-            else:
-                price_in_cents = apartment_data["debt_free_sales_price"]
-
-            price = get_euros_from_cents(price_in_cents)
-            percentage_multiplier = self.value / 100
-            apartment_installment.value = get_rounded_price(
-                price * percentage_multiplier
+        if self.unit == InstallmentUnit.EURO:
+            return apartment_installment
+        elif self.unit == InstallmentUnit.PERCENT:
+            apartment_installment.value = self._get_value_from_percentage(
+                apartment_data
             )
+            return apartment_installment
         else:
-            apartment_installment.value = self.value
+            raise ValueError(
+                f"Cannot calculate apartment installment value, {self} "
+                f'has an invalid unit "{self.unit}"'
+            )
+
+    def _get_value_from_percentage(self, apartment_data) -> Decimal:
+        if (
+            self.percentage_specifier
+            == InstallmentPercentageSpecifier.SALES_PRICE_FLEXIBLE
+        ):
+            # flexible payment's value will be populated later based on the other
+            # installments of the same apartment
+            return Decimal(0)
+        elif self.percentage_specifier == InstallmentPercentageSpecifier.SALES_PRICE:
+            price_in_cents = apartment_data["sales_price"]
+        elif (
+            self.percentage_specifier
+            == InstallmentPercentageSpecifier.DEBT_FREE_SALES_PRICE
+        ):
+            price_in_cents = apartment_data["debt_free_sales_price"]
+        elif (
+            self.percentage_specifier
+            == InstallmentPercentageSpecifier.RIGHT_OF_OCCUPANCY_PAYMENT
+        ):
+            price_in_cents = apartment_data["right_of_occupancy_payment"]
+        else:
+            raise ValueError(
+                f"Cannot calculate apartment installment value, {self} "
+                f"has an invalid percentage_specifier "
+                f'"{self.percentage_specifier}"'
+            )
```

Comment:
Black produces pretty horrible output for this. Use a temporary variable to make the code readable:
```suggestion
        ps: InstallmentPercentageSpecifier = self.percentage_specifier
        if ps == InstallmentPercentageSpecifier.SALES_PRICE_FLEXIBLE:
            # flexible payment's value will be populated later based on the other
            # installments of the same apartment
            return Decimal(0)
        elif ps == InstallmentPercentageSpecifier.SALES_PRICE:
            price_in_cents = apartment_data["sales_price"]
        elif ps == InstallmentPercentageSpecifier.DEBT_FREE_SALES_PRICE:
            price_in_cents = apartment_data["debt_free_sales_price"]
        elif ps == InstallmentPercentageSpecifier.RIGHT_OF_OCCUPANCY_PAYMENT:
            price_in_cents = apartment_data["right_of_occupancy_payment"]
        else:
            assert_never(ps)
```

Also replaced the ValueError with `assert_never` which is designed exactly for this kind of purpose. It should be imported as `from typing_extensions import assert_never` in this project. (Would be in `typing` since Python 3.11.)
https://typing-extensions.readthedocs.io/en/latest/index.html#assert_never